src/mono/wasm/runtime/loader/assets.ts

```diff
@@ -314,21 +374,44 @@ function prepareAssets(containedInSnapshotAssets: AssetEntryInternal[], alwaysLo
         }
     }
 
+    // FIXME: should we also load Net7 backward compatible `config.configs` in a same way ?
     if (config.appsettings) {
         for (let i = 0; i < config.appsettings.length; i++) {
             const configUrl = config.appsettings[i];
             const configFileName = fileName(configUrl);
             if (configFileName === "appsettings.json" || configFileName === `appsettings.${config.applicationEnvironment}.json`) {
                 alwaysLoadedAssets.push({
-                    name: configFileName,
-                    resolvedUrl: appendUniqueQuery(loaderHelpers.locateFile(configUrl), "vfs"),
-                    behavior: "vfs"
+                    name: configUrl,
+                    behavior: "vfs",
+                    // TODO what should be the virtualPath ?
+                    noCache: true,
+                    useCredentials: true
                 });
             }
+            // FIXME: why are not loading all the other named files in appsettings ? https://github.com/dotnet/runtime/issues/89861
         }
     }
 
-    config.assets = [...containedInSnapshotAssets, ...alwaysLoadedAssets];
+    config.assets = [...containedInSnapshotAssets, ...alwaysLoadedAssets, ...modulesAssets];
+}
+
+export function prepareAssetsWorker() {
+    const config = loaderHelpers.config;
+    mono_assert(config.assets, "config.assets must be defined");
+
+    for (const asset of config.assets) {
+        switch (asset.behavior) {
```

Comment:
Could you unify this with the switch from prepareAssets, in a single helper function they both use?